http-api/src/v1/projects.rs

```diff
@@ -720,3 +720,259 @@ fn resolve_revisions(
+    const COMMIT_MSG: &str = "say hi to bob";
+    const COMMIT_FILE_NAME: &str = "HI";
+    const COMMIT_FILE_CONTENT: &str = "Hi Bob";
+    const COMMIT_README_CONTENT: &str = "This is a readme";
+
+    #[tokio::test]
+    async fn test_projects_root_route() {
+        let (profile, signer, _, _) = setup::profile();
+        let ctx = Context::new(profile.paths().to_owned(), signer, THEME.to_string());
+        let app = router(ctx);
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .uri("/projects")
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+        let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        let body: Value = serde_json::from_slice(&body).unwrap();
+
+        assert_eq!(body[0]["name"], PROJECT_NAME);
+        assert_eq!(body[1], Value::Null);
+    }
+
+    #[tokio::test]
+    async fn test_project_route() {
+        let (profile, signer, project, _) = setup::profile();
+        let ctx = Context::new(profile.paths().to_owned(), signer, THEME.to_string());
+        let app = router(ctx);
+        let response = app
+            .clone()
+            .oneshot(
+                Request::builder()
+                    .uri(format!("/projects/{}", PROJECT_NAME))
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+        let alias_body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        let body: Value = serde_json::from_slice(&alias_body).unwrap();
+
+        assert_eq!(body["name"], "nakamoto");
+
+        let urn = body["urn"].as_str().unwrap();
+        assert_eq!(project.urn().to_string(), urn);
+
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .uri(format!("/projects/{}", urn))
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+        let urn_body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        assert_eq!(alias_body, urn_body);
+    }
+
+    #[tokio::test]
+    async fn test_commits_route() {
+        let (profile, signer, project, head) = setup::profile();
+        let ctx = Context::new(profile.paths().to_owned(), signer, THEME.to_string());
+        let app = router(ctx);
+        let response = app
+            .clone()
+            .oneshot(
+                Request::builder()
+                    .uri(format!("/projects/{}/commits", project.urn()))
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::FOUND);
```

Comment:
We're getting `StatusCode::FOUND` which is _somewhat_ not consistent with others.

Reply:
Yeah that's when a `parent` is not passed. Don't remember why we handled it like this though.